Feature/unassign asset

Ahmed-Eldera opened the pull request.

Ahmed-Eldera committed:
intial api without testing

Ahmed-Eldera committed:
working with Integration tests

Nourhan-Hesham commented on src/test/java/org/orange/oie/internship2025/assetmanagementsystem/controller/AssetControllerUnassignTest.java:
```diff
@@ -0,0 +1,95 @@
+
+
+@Transactional
+class AssetControllerUnassignTest extends AbstractIntegrationTest {
```
we don't add file per api we add file per controller so rename into AssetAssignmentControllerTest

Nourhan-Hesham commented on src/main/java/org/orange/oie/internship2025/assetmanagementsystem/controller/AssetAssignmentController.java:
```diff
@@ -11,15 +13,24 @@
+        this.assetAssignmentService = assetAssignmentService;
     }
+
     @PostMapping()
```
delete this controller and add this 2 apis into assetController

Ahmed-Eldera committed:
tests work in progress (teardown problem)

# Conflicts:
#	src/main/java/org/orange/oie/internship2025/assetmanagementsystem/controller/AssetController.java
#	src/test/java/org/orange/oie/internship2025/assetmanagementsystem/controller/AssetControllerTest.java

Ahmed-Eldera committed:
refactored assign and unAssign into the asset controller

Zeyad-F16 committed:
validation for serial number

Zeyad-F16 committed:
rebased
